src/main/java/seedu/duke/Parser.java

```diff
@@ -0,0 +1,151 @@
+
+        if (userInput.split(command).length < 2
+                || userInput.split(command)[1].equals(" ")) {
+            throw new DietException("☹ Error! Missing command parameters!");
+        } else {
+            switch (command) {
+            case COMMAND_NAME:
+                return userInput.split(" ")[1];
+            case COMMAND_CALCULATE:
+                for (String param: PARAM_CALCULATE) {
+                    if (userInput.contains(param)) {
+                        return userInput.split(" ")[1];
+                    }
+                }
+                throw new DietException("☹ Oops! Incorrect nutrient type");
+            case COMMAND_ADD:
+                if (!userInput.contains("n/") || !userInput.contains("x/") || !userInput.contains("k/")) {
+                    throw new DietException("☹ Oh no... Missing or incorrect add statement");
+                }
+                return userInput.substring(userInput.indexOf(' ') + 1);
+            case COMMAND_INFO:
+                if (!Arrays.asList(input).containsAll(Arrays.asList(PARAM_INFO))) {
+                    throw new DietException("☹ Oh no... Missing or incorrect info statement");
+                }
+                return userInput.substring(userInput.indexOf(' ') + 1);
+            default:
+                return null;
+            }
+        }
+    }
+
+    /**
+     * Returns the index after the command of a user input, e.g. delete 3.
+     * @param userInput user input.
+     * @return index part of the user input.
+     * @throws DietException when the user input is of a wrong format.
+     */
+    private static int getCommandIndex(String userInput) throws DietException {
+        String command = getCommand(userInput);
+
+        if (userInput.split(command).length < 2 || userInput.split(command)[1].equals(" ")) {
+            throw new DietException("☹ OOPS!!! Missing index of duke.task!");
+        }
+        try {
+            return Integer.parseInt(userInput.split(" ")[1]) - 1;
+        } catch (NumberFormatException e) {
+            throw new DietException("☹ OOPS!!! No integer index detected!");
+        }
+    }
+
+    /**
+     * Makes sense of the user input and carries out the functions according to the command given.
+     * @param userInput user input.
+     * @throws DietException when the program does not recognize the command given.
+     */
+    public static void parse(String userInput, Manager manager, Ui ui) throws DietException {
+        Calculator calculator = manager.getCalculator();
+        switch (getCommand(userInput)) {
+        case COMMAND_NAME:
+            manager.setName(getCommandParam(userInput));
+            ui.printAskForUserInfoMessage(manager.getName())
+            return;
+        case COMMAND_EXIT:
+            ui.printExitMessage(manager.getName());
+            DietBook.isExit = true;
+            return;
+        case COMMAND_LIST:
+            ui.printFoodList(manager.getFoodList());
```

Comment:
I am not sure if this will be an error, however, I have changed my printFoodList input parameter to a String instead of  a FoodList, maybe you would like to add in `.toString()` behind the `getFoodLis()`